swagger_templates/python/api_client.mustache

```diff
@@ -0,0 +1,611 @@
+
+    Swagger generic API client. This client handles the client-
+    server communication, and is invariant across implementations. Specifics of
+    the methods and models for each application are generated from the Swagger
+    templates.
+
+    NOTE: This class is auto generated by the swagger code generator program.
+    Ref: https://github.com/swagger-api/swagger-codegen
+    Do not edit the class manually.
+
+    :param host: The base path for the server to call.
+    :param header_name: a header to pass when making calls to the API.
+    :param header_value: a header value to pass when making calls to the API.
+    """
+    def __init__(self, host=None, header_name=None, header_value=None, cookie=None):
+
+        """
+        Constructor of the class.
+        """
+        self.rest_client = RESTClientObject()
+        self.default_headers = {}
+        if header_name is not None:
+            self.default_headers[header_name] = header_value
+        if host is None:
+            self.host = Configuration().host
+        else:
+            self.host = host
+        self.cookie = cookie
+        # Set default User-Agent.
+        self.user_agent = '{{#httpUserAgent}}{{{.}}}{{/httpUserAgent}}{{^httpUserAgent}}Swagger-Codegen/{{{packageVersion}}}/python{{/httpUserAgent}}'
+
+    @property
+    def user_agent(self):
+        """
+        Gets user agent.
+        """
+        return self.default_headers['User-Agent']
+
+    @user_agent.setter
+    def user_agent(self, value):
+        """
+        Sets user agent.
+        """
+        self.default_headers['User-Agent'] = value
+
+    def set_default_header(self, header_name, header_value):
+        self.default_headers[header_name] = header_value
+
+    def __call_api(self, resource_path, method,
+                   path_params=None, query_params=None, header_params=None,
+                   body=None, post_params=None, files=None,
+                   response_type=None, auth_settings=None, callback=None,
+                   _return_http_data_only=None, collection_formats=None):
+
+        # header parameters
+        header_params = header_params or {}
+        header_params.update(self.default_headers)
+        if self.cookie:
+            header_params['Cookie'] = self.cookie
+        if header_params:
+            header_params = self.sanitize_for_serialization(header_params)
+            header_params = dict(self.parameters_to_tuples(header_params,
+                                                           collection_formats))
+
+        # path parameters
+        if path_params:
+            path_params = self.sanitize_for_serialization(path_params)
+            path_params = self.parameters_to_tuples(path_params,
+                                                    collection_formats)
+            for k, v in path_params:
+                v_str = str(v)
+                # check if it is already url encoded
+                if v_str.find('%') == -1:
```

Comment:
I guess it would still fail if for some reason a quarantine path happened to contain a %2F... I'm now thinking that instead of doing the quote_plus in the api.mustache, instead it should just tag the value with some string that would be highly unlikely to ever occur in a quarantine path. Perhaps something like: